### PR TITLE
DOT-1487 automated tests

### PR DESCRIPTION
**Pull request by edwh**
(Wrong branch name)

Removed some code I think is dead, and added some more tests.

The permissions model enforced by UserController@profile looks wrong to me, but I've added tests for the current behaviour.

### CODE REVIEW

File: tests/Feature/Users/ProfileTest.php [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        // When logged out should throw an exception.
+        try {
+            $response = $this->get('/profile');
+            $this->assertFalse(TRUE);
+        } catch (AuthenticationException $e) {
+            // Success case.
+        }
+
+        // When logged in should be able to see.
+        // TODO I'm not convinced that viewing /profile is ever reachable, though /profile/id is.
```
Comment (ngm) [inline tests/Feature/Users/ProfileTest.php:30]: /profile with no id shows you your own profile, I believe.

File: tests/Feature/Users/ProfileTest.php [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+        // When logged out should throw an exception.
+        try {
+            $response = $this->post('/user/edit/' . $user1->id, $editdata);
+            $this->assertFalse(TRUE);
+        } catch (AuthenticationException $e) {
+            // Success case.
+        }
+
+        // TODO These are the behaviours as coded, but they seem weird.
```
Comment (ngm) [inline tests/Feature/Users/ProfileTest.php:64]: Agreed - I think right now, it should be only that a user can act on themselves.  Or an admin can act on anyone.

With a possible future addendum that a network coordinator could work on users in their network.

**Review by ngm — APPROVED** [review]

File: tests/Feature/Users/ProfileTest.php (hunk 1 above)
Comment (edwh) [reply tests/Feature/Users/ProfileTest.php:30]: But do we link to it anywhere?

File: tests/Feature/Users/ProfileTest.php (hunk 2 above)
Comment (edwh) [reply tests/Feature/Users/ProfileTest.php:64]: Will raise bug as agreed.

File: tests/Feature/Users/ProfileTest.php (hunk 1 above)
Comment (ngm) [reply tests/Feature/Users/ProfileTest.php:30]: Ah, right - at present, I don't know actually!

The only thing I recall related to this, was allowing for the ability to link from elsewhere, e.g. Talk, or the Wiki, to the user's profile page, without requiring knowing their id in Restarters.

**Pull request merged**